api/src/proxy/DynamoDbProxy.ts

```diff
@@ -0,0 +1,71 @@
+import { DynamoDBClient, GetItemCommand, PutItemCommand, DeleteItemCommand, ScanCommand, AttributeValue } from "@aws-sdk/client-dynamodb";
+import { Recipe } from "../model/Recipe";
+
+export class DynamoDbProxy {
+    private client: DynamoDBClient;
+    private recipeTableName: string;
+    constructor(region: string) {
+        this.client = new DynamoDBClient({ region });
+        this.recipeTableName = "StorageStack-recipeTable1D8FA348-NRYDU7A77R3U";
+    }
+
+    private toDynamoRecord(recipe: Recipe): Record<string, AttributeValue> {
+        const convert = (value: any): any => {
+            if (value === null) {
+                return { NULL: true };
+            } else if (typeof value === "string") {
+                return { S: value };
+            } else if (typeof value === "number") {
+                return { N: value.toString() };
+            } else if (typeof value === "boolean") {
+                return { BOOL: value };
+            } else if (Array.isArray(value)) {
+                return { L: value.map(convert) };
+            } else if (typeof value === "object") {
+                return { M: this.toDynamoRecord(value) };
+            } else {
+                throw new Error(`Unsupported value type: ${typeof value}`);
+            }
+        };
```

Comment:
This convert can prob be pulled out into a separate method so that we can re-use it for different things we might want to write to Dynamo